File: solutions/Python/14.LongestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[
                0
            ]  # In case there is only one value so it returns the strs directly without passing the for
        commonPrefix = ""

        for i in range(min(len(s) for s in strs)):
            char = strs[0][i]

            for j in range(len(strs)):
                if strs[j][i] != char:
                    return commonPrefix

            commonPrefix += char

        return commonPrefix  # In case there is an array with multiple ""

File: solutions/TypeScript/14.LongestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
  if (strs.length === 1) return strs[0]; // In case there is only one value so it returns the strs directly without passing the for

  let commonPrefix = "";

  for (let i: number = 0; i < Math.min(...strs.map((str) => str.length)); i++) {
    let char: string = strs[0][i];
    for (let j: number = 0; j < strs.length; j++) {
      if (strs[j][i] != char) {
        return commonPrefix;
      }
    }
    commonPrefix += char;
  }

  return commonPrefix; // In case there is an array with multiple ""
}
